sotodlib/tod_ops/detrend.py

```diff
@@ -49,28 +66,26 @@ def detrend_data(tod, method='linear', axis_name='samps',
     elif method == 'median':
         signal = signal - np.median(signal, axis=-1)[...,None]
     elif method == 'linear':
-        x = np.linspace(0, 1, n_samps)
+        x = np.linspace(0, 1, n_samps, dtype=dtype_in)
         count = max(1, min(count, signal.shape[-1] // 2))
-        slopes = signal[...,-count:].mean(axis=-1)-signal[...,:count].mean(axis=-1)
-        signal = signal - slopes[...,None] * x
+        slopes = signal[...,-count:].mean(axis=-1,dtype=dtype_in)-signal[...,:count].mean(axis=-1,dtype=dtype_in)
```

Comment:
Please flake8 this line... the suppression of spaces has not, alas, led to it all being readable on a single line.

Other lines below have arithmetic spacing violations too.